src/entities/fractions/price.rs

```diff
@@ -140,48 +147,56 @@ mod test {
     #[test]
     fn test_constructor_array_format_works() {
         let price = Price::new(TOKEN0.clone(), TOKEN1.clone(), 1, 54321);
-        assert_eq!(price.to_significant(5, Rounding::RoundDown), "54321");
-        assert!(price.meta.base_currency.equals(&TOKEN0.clone()));
-        assert!(price.meta.quote_currency.equals(&TOKEN1.clone()));
+        assert_eq!(
+            price.to_significant(5, Rounding::RoundDown).unwrap(),
+            "54321"
+        );
+        assert!(price.clone().meta.base_currency.equals(&TOKEN0.clone()));
+        assert!(price.clone().meta.quote_currency.equals(&TOKEN1.clone()));
```

Comment:
`Price` is `Clone` but `&Price` is `Copy`. `fn meta(&self)` takes a reference of `price` which is copied automatically. Please don't make unnecessary clones.
https://doc.rust-lang.org/std/marker/trait.Copy.html

Reply:
oh ,the tests yeah I must have implemented the clones when I was handling something, I will clean all the test during the next pr, probably today